modules/js-and-css/speculation-rules/readme.txt

```diff
@@ -0,0 +1,84 @@
+=== Speculation Rules ===
+
+Contributors:      wordpressdotorg
+Requires at least: 6.3
+Tested up to:      6.4
+Requires PHP:      7.0
+Stable tag:        1.0.0
+License:           GPLv2 or later
+License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+Tags:              performance, javascript, speculation rules, prerender, prefetch
+
+Uses the Speculation Rules API to prerender linked URLs upon hover.
+
+== Description ==
+
+This plugin adds support for the [Speculation Rules API](https://developer.mozilla.org/en-US/docs/Web/API/Speculation_Rules_API), which allows defining rules by which certain URLs are dynamically prefetched or prerendered based on user interaction.
+
+[See the Speculation Rules WICG specification draft.](https://wicg.github.io/nav-speculation/speculation-rules.html)
+
+By default, the plugin is configured to prerender WordPress frontend URLs when the user hovers over a relevant link. This can be customized via the "Speculation Rules" section under _Settings > Reading_.
+
+A filter can be used to exclude certain URL paths from being eligible for prefetching and prerendering (see FAQ section). Alternatively, you can add the 'no-prerender' CSS class to any link (`<a>` tag) that should not be prerendered.
+
+= Browser support =
+
+The Speculation Rules API is a new web API and currently requires using Chrome 121+.
+
+Other browsers will not see any adverse effects, however the feature will not work for those clients.
+
+* [Browser support for the Speculation Rules API in general](https://caniuse.com/mdn-html_elements_script_type_speculationrules)
+* [Information on document rules syntax support used by the plugin](https://developer.chrome.com/blog/chrome-121-beta#speculation_rules_api)
+
+== Installation ==
+
+= Installation from within WordPress =
+
+1. Visit **Plugins > Add New**.
+2. Search for **Speculation Rules**.
+3. Install and activate the **Speculation Rules** plugin.
+
+= Manual installation =
+
+1. Upload the entire `speculation-rules` folder to the `/wp-content/plugins/` directory.
+2. Visit **Plugins**.
+3. Activate the **Speculation Rules** plugin.
+
+== Frequently Asked Questions ==
+
+= How can I prevent certain URLs from being prefetched and prerendered? =
+
+Not every URL can be reasonably prerendered. Prerendering static content is typically reliable, however prerendering dynamic content, such as a shopping cart or a checkout page, can lead to issues. You can prevent certain URL paths from being prefetched or prerendered by using the `plsr_speculation_rules_href_exclude_paths` filter.
```

Comment:
This example feels slightly confusing to me.

Content that is server-side dynamic can cause surprises with either prefetching or prerendering, if the user is likely to noticeably change the state between preloading and navigation.

Prerendering additionally causes client-side code to run before navigation, which might cause surprises if there is JavaScript that does things that shouldn't happen yet.

Stuff like a shopping cart seems more likely to be an example of the former. Right?

Reply:
Makes sense. Do you have another suggestion for an example that could be used here?

Reply:
Logout link?

Reply:
Thanks, I'll update this and try to clarify.